### DISCUSSION TITLE
Behavioral tracking

### DISCUSSION THREAD

**Discussion by jesusmartin-cortecero** [petersenpeter/CellExplorer#74] [Q&A] [answered]
Hi Peter. 

Is CellExplorer Compatible with Deeplabcut?

**Comment by petersenpeter** [comment]
Hi Jesus

To what extent do you mean compatible? If you save the output from DeepLabCut to a behavior struct, then yes:
https://cellexplorer.org/datastructure/data-structure-and-format/#behavior

**Comment by petersenpeter** [comment]
Here is a tutorial on working with behavior data in CellExplorer: 
https://cellexplorer.org/tutorials/behavior-tutorial/

**Comment by jesusmartin-cortecero** [comment]
In the tutorial i see that i need OptiTrack files, i have behavioral data analyzed with deeplabcut, and i was wondering if i can import this data to cellExplorer.

**Comment by petersenpeter** [comment]
You don't need optitrack files. The tutorial just captures how to import data from that format. If you have position data from DeepLabCut you create a behavior struct with that data.

**Reply by jesusmartin-cortecero** [reply]
if i understand correctly the required data is:

 optitrack.timestamps      timestamps from optitrack
%    optitrack.position.x      x-position
%    optitrack.position.y      y-position
%    optitrack.position.z      z-position
%    optitrack.speed           speed
%    optitrack.sr              samplingrate



when i use the session-Gui, i see and option to add a file
![image](https://user-images.githubusercontent.com/53047456/195081120-43d26bf9-d4f3-4f5f-9e2f-e088356e0d20.png)
should i add there the mat structure that contains the data mentioned above? 

thank you very much in advance

**Comment by petersenpeter** [comment, marked as answer]
The structure you suggested is correct. You don't need to use the GUI - this section is for describing your raw data (could be used to describe the DeepLabCut files). Save the behavior structure with an appropriate name, it could also be:
```m
deeplabcut.timestamps = 1:100; % In seconds
deeplabcut.position.x = 1:100
deeplabcut.position.y = 1:00
deeplabcut.position.z = 1:100
deeplabcut.sr = 1;  %1 Hz
```
Not all fields are required. Then save the Matlab structure as basename.deeplabcut.behavior.mat
Then create a firingratemap structure as described in the [tutorial](https://cellexplorer.org/datastructure/data-structure-and-format/#firing-rate-maps).
Finally, rerun ProcessCellMetrics for that session, and it will import the rate maps to the cell_metrics, and you can view them in CellExplorer.

**Reply by jesusmartin-cortecero** [reply]
This is very useful, many thanks
